src/utils/hdfs_mods.py

```diff
@@ -169,185 +170,12 @@ def hdfs_write_feather(filepath, df):
-    """
-    command = ["hadoop", "fs", "-test", "-f", path]
-    return _perform(command)
-
-
-def hdfs_read_header(path: str):
-    """
-    Reads the first line of a file on HDFS
-    """
-    return _perform(
-        f"hadoop fs -cat {path} | head -1",
-        shell=True,
-        str_output=True,
-        ignore_error=True,
-    )
-
-
-def hdfs_write_string_to_file(content: bytes, path: str):
-    """
-    Writes a string into the specified file path
-    """
-    _write_string_to_file = subprocess.Popen(
-        f"hadoop fs -put - {path}", stdin=subprocess.PIPE, shell=True
-    )
-    return _write_string_to_file.communicate(content)
-
-
-def hdfs_copy_file(src_path: str, dst_path: str):
-    """
-    Copy a file from one location to another. Uses 'hadoop fs -cp'.
-    """
-    command = ["hadoop", "fs", "-cp", src_path, dst_path]
-    return _perform(command)
-
-
-def hdfs_move_file(src_path: str, dst_path: str):
-    """
-    Move a file from one location to another. Uses 'hadoop fs -mv'.
-    """
-    command = ["hadoop", "fs", "-mv", src_path, dst_path]
-    return _perform(command)
-
-
-def hdfs_list_files(path: str, ext: str = None):
-    """
-    List files in a directory. Uses 'hadoop fs -ls'.
-    """
-    # Forming the command line command and executing
-    command = ["hadoop", "fs", "-ls", path]
-    files_as_str = _perform(command, str_output=True)
-
-    # Breaking up the returned string, and stripping down to just paths of files
-    file_paths = [line.split()[-1] for line in files_as_str.strip().split("\n")[1:]]
-
-    # Getting just the filename (including ext) alone
-    file_names_in_dir = [os.path.basename(path) for path in file_paths]
-
-    # Filtering the files to just those with the required extension
-    if ext:
-        ext = f".{ext}"
-        file_names_in_dir = [
-            file for file in file_names_in_dir if os.path.splitext(file)[1] == ext
-        ]
-
-    return file_names_in_dir
-
-
-def hdfs_search_file(dir_path, ending):
-    """Find a file in a directory with a specific ending using grep and hadoop fs -ls.
+@time_logger_wrap
+def hdfs_read_feather(filepath: str) -> pd.DataFrame:
+    """Function to read feather file from HDFS"""
+    with hdfs.open(filepath, "rb") as file:
+        df = pd.read_feather(file)
+    # Check log written to feather
+    hdfs_logger.info(f"Dataframe read from {filepath} as feather file")
 
-    Args:
-        dir_path (_type_): _description_
-        ending (_type_): _description_
-    """
-    target_file = _perform(
-        f"hadoop fs -ls {dir_path} | grep {ending}",
-        shell=True,
-        str_output=True,
-        ignore_error=True,
-    )
-
-    # Handle case where file does not exist
-    if not target_file:
-        raise FileNotFoundError(
-            f"File with ending {ending} does not exist in {dir_path}"
-        )
-
-    # Return file path + name
-    target_file = target_file.split()[-1]
-
-    return target_file
```

Comment:
@jwestw has this code been removed by accident? Looks like it's still needed in the `export_files` module.

Reply:
I have tried to updated this and the below with new commits, let me know if it doesn't look right!